src/Aspire.Hosting/Backchannel/BackchannelDataTypes.cs

```diff
@@ -81,6 +81,11 @@ internal sealed class PublishingActivityData
     /// </summary>
     public required string StatusText { get; init; }
 
+    /// <summary>
+    /// Gets the completion state of the publishing activity.
+    /// </summary>
+    public string CompletionState { get; init; } = "InProgress";
```

Comment:
Make the other properties IsComplete, IsError, IsWarning , computed based in CompletionState